### PR TITLE
[native bridge move] - fix limiter error handling

### PR DESCRIPTION
**Pull request by patrickkuo**
## Description 

added Result struct to pass error info back to call site.

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

**Review by longbowlu — COMMENTED** [review]
If the motivation is to distinguish `TokenTransferAlreadyClaimed` and `TokenTransferLimitExceed`, can we instead emit the events in `fun claim_token_internal`, which should be much simpler than this change?

**Comment by patrickkuo** [comment]
> If the motivation is to distinguish `TokenTransferAlreadyClaimed` and `TokenTransferLimitExceed`, can we instead emit the events in `fun claim_token_internal`, which should be much simpler than this change?

we are not emitting any events when we call `claim_token` which will fail fast

**Comment by patrickkuo** [comment]
simplified the changes, @longbowlu thanks for pointing out it's ok to emit events for both function.

File: crates/sui-framework/packages/bridge/sources/bridge.move [hunk 1]
```diff
@@ -322,22 +326,14 @@ module bridge::bridge {
         ctx: &mut TxContext,
     ) {
         let (token, owner) = claim_token_internal<T>(clock, bridge, source_chain, bridge_seq_num, ctx);
-        if (token.is_none()) {
+        if (token.is_some()) {
+            transfer::public_transfer(
+                token.destroy_some(),
+            owner
+            )
+        }else{
```
Comment (longbowlu) [inline crates/sui-framework/packages/bridge/sources/bridge.move:334]: format

**Review by longbowlu — APPROVED** [review]
thanks! plz fix format

**Pull request merged**